### ISSUE TITLE
Define Abstraction for CPU

### ISSUE THREAD

**Issue by alex0112** [alex0112/abacus#8]
# CPU Abstraction

## Location
The abstraction for the CPU can currently be found in `src/cpu.py`

## High Level Description
The CPU is an object that has a special function called `process(...)`  That function takes in an `Opcode` object, a `Memory` object and an `IODevice`. `process` is the utility function to take the next piece of code in memory and do whatever needs to happen with it. This is likely the most sophisticated class in the project and will have a lot of other tasks associated with it, one for each opcode.

### Example
This task would be complete when another programmer can import the module create/use the object like:

```python
io_device = IODevice()
mem = Memory()
cpu = CPU()

cpu.process(opcode, mem, io_device) ## => This should modify the memory according to the opcode and/or write out to the display with the IODevice
```

### Potential Sub-Tasks

- [ ] Write a unit test for each of these cases
- [x] Ensure that the `process` function behaves as expected
- [x] The CPU will need to have logic implemented so that it knows which function to call when it receives each specific opcode
- [ ] Ensure that the accumulator functions as expected
- [ ] Mark this task complete once each individual task for each opcode has been completed and merged

**Comment by alex0112** [comment]
This is a parent issue which will be marked as complete once the logic and tests for each individual opcode is in place.

**Comment by alex0112** [comment]
Looking ahead it seems likely that for the branch logic to be implemented the CPU will also need to keep track of what piece of memory it's currently at. 

```
    BRANCH = 40 Branch to a specific location in memory
    BRANCHNEG = 41 Branch to a specific location in memory if the accumulator is negative.
    BRANCHZERO = 42 Branch to a specific location in memory if the accumulator is zero.
    HALT = 43 Stop the program
```

Which won't necessarily play nicely with the current design, but that shouldn't be too bad to fix. It might be required to either pass in memory address to the `process` function, or else pass in the entire memory to the cpu object and allow it to keep track of the current memory address.

**Comment by alex0112** [comment]
Much of this was addressed in #21. We still need the opcodes to be complete to close this out.

**Comment by alex0112** [comment]
This was finshed after milestone 1. Forgot to close
